### PR TITLE
Components context

### PR DESCRIPTION
**Pull request by orius123**
Related https://github.com/descope/etc/issues/3733

### CODE REVIEW

File: packages/web-component/src/lib/descope-wc/BaseDescopeWc.ts [hunk 1]
```diff
@@ -82,6 +82,8 @@ class BaseDescopeWc extends HTMLElement {
 
   #debugState = new State<DebugState>();
 
+  componentsContext = {};
```
Comment (nirgur) [inline packages/web-component/src/lib/descope-wc/BaseDescopeWc.ts:85]: I think we should making it private by adding a hash and add a getter fn
WDYT?

File: packages/web-component/src/lib/descope-wc/BaseDescopeWc.ts [hunk 2]
```diff
@@ -528,6 +535,11 @@ class BaseDescopeWc extends HTMLElement {
       // we want to update the state when user clicks on back in the browser
       window.addEventListener('popstate', this.#eventsCbRefs.popstate);
 
+      window.addEventListener(
```
Comment (nirgur) [inline packages/web-component/src/lib/descope-wc/BaseDescopeWc.ts:538]: Worth adding a comment about the mixin that dispatch this event

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 3]
```diff
@@ -732,10 +732,12 @@ class DescopeWc extends BaseDescopeWc {
 
       const formData = await this.#getFormData();
       const eleDescopeAttrs = getElementDescopeAttributes(submitter);
+      const contextArgs = this.componentsContext;
 
       const actionArgs = {
         ...eleDescopeAttrs,
         ...formData,
+        ...contextArgs,
```
Comment (nirgur) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:740]: We need to think what is the correct order here

**Review by nirgur — COMMENTED** [review]
Looks good,
Added a few small comments

**Comment by orius123** [comment]
@nirgur I fixed you comments, lmk if you want to think about the order thing together

**Comment by nirgur** [comment]
Let's wait with it until I fix all tests
Hopefully it will be today

**Comment by orius123** [comment]
@nirgur can we merge this one?

**Comment by nirgur** [comment]
> @nirgur can we merge this one?

👍

**Pull request merged**